src/client/Microsoft.Identity.Client/ManagedIdentity/Csr.cs

```diff
@@ -0,0 +1,439 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client.ManagedIdentity
+{
+    internal class Csr
+    {
+        public string Pem { get; }
+
+        public Csr(string pem)
+        {
+            Pem = pem ?? throw new ArgumentNullException(nameof(pem));
+        }
+
+        /// <summary>
+        /// Generates a CSR for the given client, tenant, and CUID info.
+        /// </summary>
+        /// <param name="clientId">Managed Identity client_id.</param>
+        /// <param name="tenantId">AAD tenant_id.</param>
+        /// <param name="cuid">CuidInfo object containing required VMID and optional VMSSID.</param>
```

Comment:
VMSSID isn't used anywhere? except in tests?

Reply:
discussion to clear this up occurred offline